gravitee-apim-console-webui/src/management/integrations/integrations.component.ts

```diff
@@ -36,44 +38,68 @@ export class IntegrationsComponent implements OnInit {
   public isLoading: boolean = false;
   public integrations: Integration[] = [];
   public displayedColumns: string[] = ['name', 'owner', 'provider', 'agent', 'action'];
-
+  public isFreeTier: boolean = false;
   public filters: GioTableWrapperFilters = {
     pagination: { index: 1, size: 10 },
     searchTerm: '',
   };
   public nbTotalInstances = this.integrations.length;
-
   private filters$ = new BehaviorSubject<GioTableWrapperFilters>(this.filters);
 
   constructor(
     private integrationsService: IntegrationsService,
     private snackBarService: SnackBarService,
+    private licenseService: GioLicenseService,
   ) {}
 
   ngOnInit(): void {
-    this.filters$
+    this.licenseService
+      .getLicense$()
       .pipe(
-        distinctUntilChanged(isEqual),
-        switchMap((filters: GioTableWrapperFilters) => {
-          this.isLoading = true;
-          return this.integrationsService.getIntegrations(filters.pagination.index, filters.pagination.size);
+        mergeMap((license: License): Observable<null> | Observable<IntegrationResponse> => {
+          if (license.isExpired || license.tier === 'oss') {
+            this.isFreeTier = true;
+            return of(null);
+          }
+          return this.initFilters();
         }),
-        catchError((_) => {
+        catchError(({ error }) => {
           this.isLoading = false;
-          this.snackBarService.error('Something went wrong!');
+          this.snackBarService.error(error.message);
           return EMPTY;
         }),
         takeUntilDestroyed(this.destroyRef),
       )
-      .subscribe((response: IntegrationResponse): void => {
-        this.nbTotalInstances = response.pagination.totalCount;
-        this.integrations = response.data;
+      .subscribe((response: IntegrationResponse | null): void => {
         this.isLoading = false;
+        if (response) {
+          this.nbTotalInstances = response.pagination.totalCount;
+          this.integrations = response.data;
+        }
       });
   }
 
+  initFilters(): Observable<IntegrationResponse> {
+    return this.filters$.pipe(
+      distinctUntilChanged(isEqual),
+      switchMap((filters: GioTableWrapperFilters) => {
+        this.isLoading = true;
```

Comment:
Non-blocking comment: I would move the `this.isLoading = true` into the logic in `ngOnInit` because it can be difficult to debug if we set `isLoading` in sub-methods.